pkg/pxc/app/pmm.go

```diff
@@ -161,14 +161,16 @@ func pmmAgentEnvs(pmmServerHost, pmmServerUser, secrets string) []corev1.EnvVar
 func PMMAgentScript(dbType string) []corev1.EnvVar {
 	pmmServerArgs := " $(PMM_ADMIN_CUSTOM_PARAMS) --skip-connection-check --metrics-mode=push "
 	pmmServerArgs += " --username=$(DB_USER) --password=$(DB_PASSWORD) --cluster=$(CLUSTER_NAME) "
-	pmmServerArgs += " --service-name=$(PMM_AGENT_SETUP_NODE_NAME) --host=$(POD_NAME) --port=$(DB_PORT) "
 	if dbType == "mysql" {
-		pmmServerArgs += "$(DB_ARGS)"
+		pmmServerArgs += "$(DB_ARGS) "
+	}
+	if dbType != "haproxy" {
+		pmmServerArgs += "--host=$(POD_NAME) --port=$(DB_PORT)"
 	}
 	return []corev1.EnvVar{
 		{
 			Name:  "PMM_AGENT_PRERUN_SCRIPT",
-			Value: "pmm-admin status --wait=10s;\npmm-admin add $(DB_TYPE)" + pmmServerArgs + ";\npmm-admin annotate --service-name=$(PMM_AGENT_SETUP_NODE_NAME) 'Service restarted'",
+			Value: "pmm-admin status --wait=10s;\npmm-admin add $(DB_TYPE)" + pmmServerArgs + " $(PMM_AGENT_SETUP_NODE_NAME);\npmm-admin annotate --service-name=$(PMM_AGENT_SETUP_NODE_NAME) 'Service restarted'",
```

Comment:
maybe missed option
```suggestion
			Value: "pmm-admin status --wait=10s;\npmm-admin add $(DB_TYPE)" + pmmServerArgs + " --service-name=$(PMM_AGENT_SETUP_NODE_NAME);\npmm-admin annotate --service-name=$(PMM_AGENT_SETUP_NODE_NAME) 'Service restarted'",
```

Reply:
pmm-admin doesn't have `--service-name` option for `haproxy` but it seems all `add` subcommands accept the service name as a positional argument